src/test/java/net/snowflake/ingest/streaming/internal/DataValidationUtilTest.java

```diff
@@ -251,20 +253,47 @@ public void testValidateAndParseTime() {
   }
 
   @Test
-  public void testValidateAndParseTimestamp() {
+  public void testValidateAndParseTimestamp() throws ParseException {
     TimestampWrapper wrapper =
         DataValidationUtil.validateAndParseTimestamp(
             "COL", "2021-01-01T01:00:00.123+01:00", 4, UTC, false, 0);
-    assertEquals(1609459200, wrapper.getEpoch());
+    assertEquals(1609459200, wrapper.getEpochSecond());
     assertEquals(123000000, wrapper.getFraction());
     assertEquals(3600, wrapper.getTimezoneOffsetSeconds());
     assertEquals(1500, wrapper.getTimeZoneIndex());
 
     wrapper = validateAndParseTimestamp("COL", "  2021-01-01T01:00:00.123 \t\n", 9, UTC, true, 0);
-    Assert.assertEquals(1609462800, wrapper.getEpoch());
+    Assert.assertEquals(1609462800, wrapper.getEpochSecond());
     Assert.assertEquals(123000000, wrapper.getFraction());
     Assert.assertEquals(new BigInteger("1609462800123000000"), wrapper.toBinary(false));
 
+    // Test integer-stored time and scale guessing
+    SimpleDateFormat df = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS");
```

Comment:
I think we also need integration test coverage, which would test JDBC compatibility. We have a bunch of IT tests for scale guessing in [DateTimeIT.java](https://github.com/snowflakedb/snowflake-ingest-java/blob/master/src/test/java/net/snowflake/ingest/streaming/internal/datatypes/DateTimeIT.java), just not for these edge cases. Could you add tests along the lines of [these](https://github.com/snowflakedb/snowflake-ingest-java/blob/master/src/test/java/net/snowflake/ingest/streaming/internal/datatypes/DateTimeIT.java#L382)?

Reply:
added